src/sinks/datadog/traces/stats.rs

```diff
@@ -0,0 +1,532 @@
+    if parent_id_to_child_weigth.len() != 1 {
+        debug!("Didn't reliably find the root span.");
+    }
+
+    *parent_id_to_child_weigth
+        .values()
+        .next()
+        .unwrap_or_else(|| {
+            debug!("Root span was not found");
+            &1.0
+        })
+}
+
+pub(crate) fn compute_apm_stats(key: &PartitionKey, traces: &[TraceEvent]) -> StatsPayload {
+    let mut aggregator = Aggregator::new();
+    traces.iter().for_each(|t| aggregator.handle_trace(key, t));
+    StatsPayload {
+        agent_hostname: key.hostname.clone().unwrap_or_else(|| "".to_string()),
+        agent_env: key.env.clone().unwrap_or_else(|| "".to_string()),
+        stats: aggregator.get_client_stats_payload(),
+        agent_version: key.agent_version.clone().unwrap_or_else(|| "".to_string()),
+        client_computed: false,
+    }
+}
+
+// Stats struct come from .proto and field name are set to match go
+// See this https://github.com/DataDog/datadog-agent/blob/b5bed4d/pkg/trace/pb/stats_gen.go for the reference go code
+#[derive(Debug, PartialEq, Deserialize, Serialize)]
+pub(crate) struct StatsPayload {
+    #[serde(rename = "AgentHostname")]
+    pub(crate) agent_hostname: String,
+    #[serde(rename = "AgentEnv")]
+    pub(crate) agent_env: String,
+    #[serde(rename = "Stats")]
+    pub(crate) stats: Vec<ClientStatsPayload>,
+    #[serde(rename = "AgentVersion")]
+    pub(crate) agent_version: String,
+    #[serde(rename = "ClientComputed")]
+    pub(crate) client_computed: bool,
+}
+
+#[derive(Debug, PartialEq, Deserialize, Serialize)]
+pub(crate) struct ClientStatsPayload {
+    #[serde(rename = "Hostname")]
+    pub(crate) hostname: String,
+    #[serde(rename = "Env")]
+    pub(crate) env: String,
+    #[serde(rename = "Version")]
+    pub(crate) version: String,
+    #[serde(rename = "Stats")]
+    pub(crate) stats: Vec<ClientStatsBucket>,
+    #[serde(rename = "Lang")]
+    pub(crate) lang: String,
+    #[serde(rename = "TracerVersion")]
+    pub(crate) tracer_version: String,
+    #[serde(rename = "RuntimeID")]
+    pub(crate) runtime_id: String,
+    #[serde(rename = "Sequence")]
+    pub(crate) sequence: u64,
+    #[serde(rename = "AgentAggregation")]
+    pub(crate) agent_aggregation: String,
+    #[serde(rename = "Service")]
+    pub(crate) service: String,
+    #[serde(rename = "ContainerID")]
+    pub(crate) container_id: String,
+    #[serde(rename = "Tags")]
+    pub(crate) tags: Vec<String>,
+}
+
+#[derive(Debug, PartialEq, Deserialize, Serialize)]
+pub(crate) struct ClientStatsBucket {
+    #[serde(rename = "Start")]
+    pub(crate) start: u64,
+    #[serde(rename = "Duration")]
+    pub(crate) duration: u64,
+    #[serde(rename = "Stats")]
+    pub(crate) stats: Vec<ClientGroupedStats>,
+    #[serde(rename = "AgentTimeShift")]
+    pub(crate) agent_time_shift: i64,
+}
+
+#[derive(Debug, PartialEq, Deserialize, Serialize)]
+pub(crate) struct ClientGroupedStats {
```

Comment:
```suggestion
#[derive(Debug, PartialEq, Deserialize, Serialize)]
#[serde(rename_all = "PascalCase")]
pub(crate) struct ClientGroupedStats {
```

(`http_status_code`, `db_type` and `ok_summary` still need manual renames)